code/modules/mob/living/carbon/human/species/species.dm

```diff
@@ -189,6 +192,70 @@
+		return
+
+	//Responding to a raised hand
+	if(target.flags_emote & EMOTING_ROCK_PAPER_SCISSORS && do_after(H, 5, INTERRUPT_MOVED, EMOTE_ICON_ROCK_PAPER_SCISSORS))
+		if(!(target.flags_emote & EMOTING_ROCK_PAPER_SCISSORS)) //Additional check for if the target moved or was already high fived.
+			to_chat(H, SPAN_NOTICE("Too slow!"))
+			return
+		target.flags_emote &= ~EMOTING_ROCK_PAPER_SCISSORS
+		var/static/list/game_quips = list("Rock...", "Paper...", "Scissors...", "Shoot!")
+		for(var/quip in game_quips)
+			if(!H.Adjacent(target))
+				to_chat(list(H, target), SPAN_WARNING("You need to be adjacent to eachother to play!"))
+				return
+			to_chat(list(H, target), SPAN_NOTICE(quip))
+			sleep(5)
+		var/static/list/intent_to_play = list(
+			"[INTENT_HELP]" = "random",
+			"[INTENT_DISARM]" = "scissors",
+			"[INTENT_GRAB]" = "paper",
+			"[INTENT_HARM]" = "rock"
+		)
+		var/static/list/play_to_emote = list(
+			"rock" = EMOTE_ICON_ROCK,
+			"paper" = EMOTE_ICON_PAPER,
+			"scissors" = EMOTE_ICON_SCISSORS
+		)
+		var/protagonist_plays = intent_to_play["[H.a_intent]"] == "random" ? pick("rock", "paper", "scissors") : intent_to_play["[H.a_intent]"]
+		var/antagonist_plays = intent_to_play["[target.a_intent]"] == "random" ? pick("rock", "paper", "scissors") : intent_to_play["[target.a_intent]"]
+		var/winner_text = " It's a draw!"
+		if(protagonist_plays != antagonist_plays)
+			var/static/list/what_beats_what = list("rock" = "scissors", "scissors" = "paper", "paper" = "rock")
+			if(antagonist_plays == what_beats_what[protagonist_plays])
+				winner_text = " [H] wins!"
+			else
+				winner_text = " [target] wins!"
+		H.visible_message(SPAN_NOTICE("[H] plays <b>[protagonist_plays]</b>![winner_text]"), SPAN_NOTICE("You play <b>[protagonist_plays]</b>![winner_text]"), max_distance = 5)
+		target.visible_message(SPAN_NOTICE("[target] plays <b>[antagonist_plays]</b>![winner_text]"), SPAN_NOTICE("You play <b>[antagonist_plays]</b>![winner_text]"), max_distance = 5)
+		playsound(target, "clownstep", 35, TRUE)
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/do_after, H, 8, INTERRUPT_NONE, play_to_emote[protagonist_plays])
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/do_after, target, 8, INTERRUPT_NONE, play_to_emote[antagonist_plays])
+		H.animation_attack_on(target)
+		target.animation_attack_on(H)
+		H.start_audio_emote_cooldown(5 SECONDS)
+		target.start_audio_emote_cooldown(5 SECONDS)
+		return
+
+	//Initiate high five
+	if(H.recent_audio_emote)
+		to_chat(H, "You just did an audible emote. Wait a while.")
+		return
+
+	H.visible_message(SPAN_NOTICE("[H] challenges [target] to a game of rock paper scissors!"), SPAN_NOTICE("You challenge [target] to a game of rock paper scissors!"), null, 4)
+	H.flags_emote |= EMOTING_ROCK_PAPER_SCISSORS
+	if(do_after(H, 50, INTERRUPT_ALL|INTERRUPT_EMOTE, EMOTE_ICON_ROCK_PAPER_SCISSORS) && H.flags_emote & EMOTING_ROCK_PAPER_SCISSORS)
+		to_chat(H, SPAN_NOTICE("You were left hanging!"))
+	H.flags_emote &= ~EMOTING_ROCK_PAPER_SCISSORS
```

Comment:
what's going on here

Reply:
that do_after prevents the flag from being taken away while you wait on the other guy, it's what all the other emotes do